04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,66 @@
+import Product from "../models/product.js";
+
+async function getAllProductsStatic(req, res) {
+  const products = await Product.find({ name: "vase table" });
+  res.status(200).json({ products, nbHits: products.length });
+}
+
+async function getAllProducts(req, res) {
+  const { featured, company, name, sort, fields, numericFilters } = req.query;
+  const queryObject = {};
+  if (featured) {
+    queryObject.featured = featured === "true" ? true : false;
+  }
+  if (company) {
+    queryObject.company = company;
+  }
+  if (name) {
+    queryObject.name = { $regex: name, $options: "i" };
+  }
+  let result = Product.find(queryObject);
+
+  if (sort) {
+    const sortParams = sort.split(",").join(" "); //parse incoming qParams for using at server
+    result = result.sort(sortParams);
+  } else {
+    result = result.sort("createAt");
```

Comment:
```suggestion
    result = result.sort("createdAt");
```

Make sure to test this endpoint without passing the sort param and expecting the response to have an ordered list based on the creation date.